reservation/views.py

```diff
@@ -68,3 +42,117 @@ def get_csv_file(request):
         return response
 
     return render(request, 'reservation/user_reservation.html', {'reservations': reservations})
+
+
+@login_required()
+@ratelimit(key="ip", rate="30/m", block=True)
+def select_cinema(request):
+    cinemas = MovieTheater.objects.all()
+    return render(request, 'reservation/select_cinema.html', {'cinemas': cinemas})
+
+
+def select_movie(request, cinema_pk):
+    if request.method == "GET":
+        movies_playing_time = PlayingTime.objects.filter(assigned_hall__in=Hall.objects.filter(movie_theater=cinema_pk), start_time__range=get_current_week())
```

Comment:
Try to simplify this query. See if you can use `distinct` and `values_list`. You might not even need the following if block afterwards.